.github/workflows/reusable_docker_pipeline.yml

```diff
@@ -25,33 +25,74 @@ on:
         description: 'Enable authentication for private repositories'
         type: boolean
         default: false
+      platforms:
+        required: false
+        type: string
+        description: "Target platforms for Docker build"
+        default: "linux/amd64,linux/arm64"
 
 jobs:
-  docker_build:
+  prepare-metadata:
     runs-on: ubuntu-22.04
+    outputs:
+      image-name: ${{ steps.set_image_name.outputs.IMAGE_NAME }}
     steps:
+      - name: Determine image name
+        id: set_image_name
+        run: |
+          if [ -n "${{ inputs.repoName }}" ]; then
+            echo "IMAGE_NAME=${{ inputs.repoName }}" >> $GITHUB_OUTPUT
+          else
+            echo "IMAGE_NAME=$(echo $GITHUB_REPOSITORY | cut -d '/' -f 2)" >> $GITHUB_OUTPUT
+          fi
+
+  docker_build:
+    runs-on: ${{ matrix.runner }}
+    needs: prepare-metadata
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - linux/amd64
+          - linux/arm64
+        runner:
+          - ubuntu-24.04
+          - ubuntu-24.04-arm64
+        exclude:
+        - platform: linux/amd64
+          runner: ubuntu-24.04-arm64
+        - platform: linux/arm64
+          runner: ubuntu-24.04
+    steps:
+      - name: Prepare
```

Comment:
Could this be unified with the `prepare-metadata` job?

Reply:
This step gets the `platform=${{ matrix.platform }}` which is only available in the `docker_build` job